lib/matplotlib/backends/backend_cairo.py

```diff
@@ -463,11 +475,10 @@ def print_rgba(self, fobj):
 
     def _get_printed_image_surface(self):
         width, height = self.get_width_height()
-        renderer = RendererCairo(self.figure.dpi)
```

Comment:
This used to make a new renderer, now it gets the cached one.  Why should we not be worried about this?

Reply:
I realized that I forgot to update self._renderer.dpi (now fixed), but the point is that other than the dpi, the width/height, and the surface, there's no other state on the renderer (the graphicscontext is reset between draws), and all of these always get reset manually before each draw.